mm2src/mm2_main/src/rpc.rs

```diff
@@ -175,12 +175,25 @@ fn response_from_dispatcher_error(
     response.serialize_http_response()
 }
 
-async fn process_single_request(ctx: MmArc, req: Json, client: SocketAddr) -> Result<Response<Vec<u8>>, String> {
+async fn process_single_request(ctx: MmArc, mut req: Json, client: SocketAddr) -> Result<Response<Vec<u8>>, String> {
     let local_only = ctx.conf["rpc_local_only"].as_bool().unwrap_or(true);
     if req["mmrpc"].is_null() {
-        return dispatcher_legacy::process_single_request(ctx, req, client, local_only)
-            .await
-            .map_err(|e| ERRL!("{}", e));
+        match dispatcher_legacy::process_single_request(ctx.clone(), req.clone(), client, local_only).await {
```

Comment:
Should we not first try rpc version 2 rather than legacy?
to make preference for v2, in case of identical rpc name for v2 and legacy

Reply:
This approach will cause BC on identical RPC names as they don't set `mmrpc` field.